### DISCUSSION TITLE
fastlane gym flutter flavor

### DISCUSSION THREAD

**Discussion by Milos5611** [fastlane/fastlane#21701] [Q&A]
How can I tell to `gym` to not look at `lib/main.dart` but `lib/production_main.dart`?
This is how I have it right now:
```
gym(
  configuration: "Release",
  workspace: "Runner.xcworkspace",
  scheme: "production",
  export_method: "app-store",
  skip_profile_detection: true,
  output_directory: "./build/ipa/",
  output_name: "Runner.ipa",
  include_bitcode: true,
  export_options: {
    provisioningProfiles: {
        "xxx" => "match appstore xxx"
    }
  }
)
```
And I constantly get => `❌  error: lib/main.dart: Error: Error when reading 'lib/main.dart': No such file or directory`

What I'm doing wrong? Can't wrap my head around this for quite some time.

**Comment by LalitMujalde** [comment]
If gym doesn't support a direct configuration for a different entry point file, you might need to change the file name temporarily for the build process.

For example:

Rename lib/production_main.dart to lib/main.dart.
Run gym or the build command.
Rename it back to lib/production_main.dart.

This is a workaround and might not be the most efficient solution, but it can help if the tool doesn't have an explicit option to specify the entry point file.

**Comment by niti-alzhemist** [comment]
You can create [flavor].xcconfig to override Flutter_target 
```
#include "Generated.xcconfig"
#include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"

FLUTTER_TARGET=lib/main_preprod.dart
```

And then set it in Xcode 
`Project -> Runner -> Info -> Configurations`
